core/src/main/python/wlsdeploy/util/target_configuration_helper.py

```diff
@@ -269,6 +273,11 @@ def _create_secret_name(variable_name):
     # rejoin with hyphens, remove leading and trailing hyphens from final name.
     # if empty, just return "x".
     secret = '-'.join(secret_keys).strip('-')
+    if len(secret) > secret_length:
+        secret = secret[:secret_length] + '%05d' % __long_key
+        __long_key += 1
```

Comment:
I think this will result in the secret names in the script having different numbers than the ones in the model, as this method is called again for each variable/secret name. If this fix is applied with PR #723, this is resolved.
This method is called again when the template hashes are generated for target vz, even with #723, so it may need to be revised to use the credential cache.